cmd/hub/cli/version.go

```diff
@@ -63,34 +70,32 @@ var versionCmd = &cobra.Command{
 			cmd.Message("The Hub API is running %s", apiVersion)
 		}
 
-		latest, err := checkProduction()
+		latestRelease, err := getLatestRelease()
 		if err != nil {
 			cmd.Error(err)
 			cmd.Warn("Unable to get latest release.")
 		} else {
-			current, err := semver.ParseTolerant(version)
+			current, err := semver.ParseTolerant(bi.Version)
 			if err != nil {
-				// Display warning if off production
-				cmd.Warn("Running a custom hub build. Run `%s` to install the latest release.", aurora.White("hub update").Bold())
-			} else if current.LT(latest.Version) {
+				// Display warning if not using a release
+				cmd.Warn("Running a custom hub build. Run %s to install the latest release.", aurora.White("hub update").Bold())
+				if isGovvvBuild {
+					cmd.Message("Custom build info:")
+					cmd.RenderTable(
+						[]string{},
+						[][]string{
+							{"Git Branch", bi.GitBranch},
+							{"Git State", bi.GitState},
+							{"Git Commit", bi.GitCommit},
+							{"Git Summary", bi.GitSummary},
+							{"Build Date", bi.BuildDate},
+						},
+					)
+				}
+			} else if current.LT(latestRelease.Version) {
 				// Display warning if outdated
-				cmd.Warn("There is a new hub release. Run `%s` to install %s.", aurora.White("hub update").Bold(), aurora.Cyan(latest.Version.String()))
+				cmd.Warn("There is a new hub release. Run %s to install %s.", aurora.White("hub update").Bold(), aurora.Cyan(latestRelease.Version.String()))
```

Comment:
Ok see what you think of this @andrewxhill and @sanderpick. I didn't go as far as doing custom parsing of the git summary, but I think this represents everything we talked about in today's call.

- Always print out some version string. This will be the `buildinfo.Version` for CI release builds, `buildinfo.GitSummary` for local makefile builds, or `git` for local `go install ./...` builds.
- Always print out the API version.
- If we can parse `buildinfo.Version` as a valid version, we compare it to the latest release and possibly print out the  warning message that there is a newer version available.
- If we can't parse `buildinfo.Version` as a valid version (this is the case for local builds using `make` or `go install`) we print the "running a custom build" warning.
- If we can't parse `buildinfo.Version` as a valid version AND we have `buildinfo` set because it was a `make` build,  we print out a table showing all the build information as context.

Reply:
Sounds good to me!